### PR TITLE
fix(staff): Remove idle expiration time for staff

### PR DESCRIPTION
**Pull request by schew2381**
Remove idle expiration time for staff so we don't send the session cookie back on every new staff request and reduce the max age for staff.

Django's [session modified check](https://github.com/django/django/blob/main/django/contrib/sessions/backends/base.py#L57-L67) is basic because it will always say it's modified if you add or remove a key, regardless of whether or not the value existed before and was identical. Therefore we need to incorporate that logic directly into the staff logic.

Closes https://github.com/getsentry/team-core-product-foundations/issues/249

### CODE REVIEW

File: src/sentry/auth/staff.py [hunk 1]
```diff
@@ -229,7 +213,7 @@ def _populate(self) -> None:
         if not data:
             self._set_logged_out()
         else:
-            self._set_logged_in(expires=data["exp"], token=data["tok"], user=user)
+            self._set_logged_in(_seconds_to_timestamp(data["exp"]), token=data["tok"], user=user)
```
Comment (cathteng) [inline src/sentry/auth/staff.py:216]: can we still type this as `expires=_seconds_to_timestamp(data["exp"])`

File: src/sentry/auth/staff.py [hunk 2]
```diff
@@ -179,24 +180,7 @@ def get_session_data(self, current_datetime=None):
...
-        if data["idl"] < current_datetime:
-            logger.info(
-                "staff.session-expired",
-                extra={"ip_address": request.META["REMOTE_ADDR"], "user_id": request.user.id},
-            )
-            return
-
-        try:
-            data["exp"] = datetime.fromtimestamp(float(data["exp"]), timezone.utc)
+            expires_date = _seconds_to_timestamp(data["exp"])
```
Comment (cathteng) [inline src/sentry/auth/staff.py:183]: do we have to convert this every time? could we store it in the correct format in the first place 🤔

File: src/sentry/auth/staff.py [hunk 3]
```diff
@@ -250,32 +234,35 @@ def _populate(self) -> None:
...
         # do we have a valid staff session?
         self.is_valid = True
         # is the session active? (it could be valid, but inactive)
         self._is_active, self._inactive_reason = self.is_privileged_request()
-        self.request.session[SESSION_KEY] = {
-            "exp": self.expires.strftime("%s"),
-            "idl": (current_datetime + IDLE_MAX_AGE).strftime("%s"),
+
+        session_info = {
+            "exp": expires.strftime("%s"),
```
Comment (cathteng) [inline src/sentry/auth/staff.py:252]: are we using this anywhere else in this raw form?

File: tests/sentry/auth/test_staff.py [hunk 4]
```diff
@@ -192,7 +179,31 @@ def test_login_saves_session(self):
...
+        # Mock the signed cookie in the request to match the token in the session
+        request.get_signed_cookie = Mock(return_value=request_staff.token)  # type: ignore[method-assign]
+
+        activated_staff = Staff(request)
+
+        # Staff should still be active
+        assert activated_staff.is_active
+        # The session should not be modified because the staff key in the
+        # session wasn't replaced
+        assert request.session.modified is False
```
Comment (cathteng) [inline tests/sentry/auth/test_staff.py:201]: was this modified before your changes?

**Review by cathteng — APPROVED** [review]
i'm good with this change. i would like to see the new test you wrote fail without these changes though

File: tests/sentry/auth/test_staff.py (hunk 4 above)
Comment (schew2381) [reply tests/sentry/auth/test_staff.py:201]: Yes it was always modified

File: src/sentry/auth/staff.py (hunk 3 above)
Comment (schew2381) [reply src/sentry/auth/staff.py:252]: I don't think so

File: src/sentry/auth/staff.py (hunk 2 above)
Comment (schew2381) [reply src/sentry/auth/staff.py:183]: I can try storing the datetime format itself

Comment (schew2381) [reply src/sentry/auth/staff.py]: ok jk due to serializing concerns it must be a timestamp

**Pull request merged**